packages/components/datepicker/src/components/osds-datepicker/core/controller.ts

```diff
@@ -23,10 +23,17 @@ class OdsDatepickerController {
       if (newValue === undefined || newValue === null || isNaN(newValue.getTime())) {
```

Comment:
you can add also `|| !this.validateValue(newValue)` condition and remove the last else

Reply:
Can you create a function for add context like `isValidValue`